src/ir/opt/cfold.rs

```diff
@@ -85,6 +85,8 @@ pub fn fold_cache(node: &Term, cache: &mut TermMap<Term>) -> Term {
                 match (o, c0.as_bv_opt(), c1.as_bv_opt()) {
                     (Sub, Some(a), Some(b)) => cbv(a.clone() - b.clone()),
                     (Sub, _, Some(b)) if b.uint() == &Integer::from(0) => Some(c0.clone()),
+                    // TODO: is this right?
+                    (Udiv, _, Some(b)) if b.uint() == &Integer::from(0) => Some(c0.clone()),
```

Comment:
Not quite. We're following SMT-LIB, which says this should be the maximum representable value:
> [[(bvudiv s t)]] := if bv2nat([[t]]) = 0
                       then λx:[0, m). 1
                       else nat2bv[m](bv2nat([[s]]) div bv2nat([[t]]))

See: https://smtlib.cs.uiowa.edu/theories-FixedSizeBitVectors.shtml or https://github.com/circify/circ/blob/master/src/ir/term/bv.rs#L38-L61